File: 29Items-Projects/9Web-design/4-Travel-Agency-Website/netlify/functions/health.js
const { query } = require("../../src/api/db/pool");
const { json } = require("../../src/api/utils/http");

exports.handler = async () => {
  try {
    await query("SELECT 1");
    return json(200, { status: "ok", database: "ok" });
  } catch (_error) {
    return json(503, { status: "error", database: "unavailable" });
  }
};

File: 29Items-Projects/14Rust/5-Real-Time-Chat-App/backend/src/api/health.rs
use axum::{Json, extract::State, http::StatusCode};
use serde::Serialize;

use crate::state::AppState;

#[derive(Debug, Serialize)]
pub struct HealthResponse {
    status: &'static str,
}

pub async fn live() -> Json<HealthResponse> {
    Json(HealthResponse { status: "ok" })
}

pub async fn ready(
    State(state): State<AppState>,
) -> Result<Json<HealthResponse>, (StatusCode, Json<HealthResponse>)> {
    sqlx::query_scalar::<_, i32>("SELECT 1")
        .fetch_one(&state.pool)
        .await
        .map(|_| Json(HealthResponse { status: "ready" }))
        .map_err(|error| {
            tracing::warn!(error = ?error, "readiness database check failed");
            (
                StatusCode::SERVICE_UNAVAILABLE,
                Json(HealthResponse {
                    status: "not_ready",
                }),
            )
        })
}
